core/API/CORSHandler.php

```diff
@@ -25,8 +25,22 @@ public function __construct()
 
     public function handle()
     {
+        if (empty($this->domains)) {
+            return;
+        }
+        
+        Common::sendHeader('Vary: Origin');
```

Comment:
Is there a reason for this new header? From what I understand about the header, it exists to tell http caches how to identify a resource when deciding which cached object to use. It doesn't seem related to CORS.

Reply:
> 6.4 Implementation Considerations
This section is non-normative.

Resources that wish to enable themselves to be shared with multiple Origins but do not respond uniformly with "*" must in practice generate the Access-Control-Allow-Origin header dynamically in response to every request they wish to allow. As a consequence, authors of such resources should send a Vary: Origin HTTP header or provide other appropriate control directives to prevent caching of such responses, which may be inaccurate if re-used across-origins.

https://www.w3.org/TR/cors/#resource-implementation

Reply:
I see, 👍